repositories/org_repository.go

```diff
@@ -66,3 +99,68 @@ func (r *OrgRepo) FetchOrgs(ctx context.Context, names []string) ([]OrgRecord, e
 
 	return records, nil
 }
+
+func (r *OrgRepo) FetchSpaces(ctx context.Context, organizationGUIDs, names []string) ([]SpaceRecord, error) {
+	subnamespaceAnchorList := &v1alpha2.SubnamespaceAnchorList{}
+
+	err := r.privilegedClient.List(ctx, subnamespaceAnchorList)
+	if err != nil {
+		return nil, err
+	}
+
+	orgsFilter := toMap(organizationGUIDs)
+	orgUIDs := map[string]string{}
+	for _, anchor := range subnamespaceAnchorList.Items {
+		if anchor.Namespace != r.rootNamespace {
+			continue
+		}
+
+		anchorUID := string(anchor.UID)
+		if !matchFilter(orgsFilter, anchorUID) {
+			continue
+		}
+
+		orgUIDs[anchor.Name] = anchorUID
+	}
+
+	nameFilter := toMap(names)
+	records := []SpaceRecord{}
+	for _, anchor := range subnamespaceAnchorList.Items {
+		spaceName := anchor.Labels[SpaceNameLabel]
+		if !matchFilter(nameFilter, spaceName) {
+			continue
+		}
+
+		if _, ok := orgUIDs[anchor.Namespace]; !ok {
+			continue
+		}
+
+		records = append(records, SpaceRecord{
+			Name:             spaceName,
+			GUID:             string(anchor.UID),
+			OrganizationGUID: orgUIDs[anchor.Namespace],
+			CreatedAt:        anchor.CreationTimestamp.Time,
+			UpdatedAt:        anchor.CreationTimestamp.Time,
+		})
+	}
+
+	return records, nil
+}
+
+func matchFilter(filter map[string]bool, value string) bool {
+	if len(filter) == 0 {
+		return true
+	}
+
+	_, ok := filter[value]
+	return ok
+}
+
+func toMap(elements []string) map[string]bool {
```

Comment:
Since you're never looking at the value for the map keys, you can change this type to `map[string]struct{}` to communicate that fact.

Reply:
Right, I agree empty struct is better. Done